tt-metal build is broken by commit `89cf9e2`: include not found

### Describe the bug

After this commit [89cf9e25](https://github.com/tenstorrent/tt-metal/commit/89cf9e2566ea361d03b703577d027f376204f07d) I cannot build tt-metal with the following error:
```
/usr/bin/ccache /usr/lib/ccache/clang++-17 -DBOOST_CONTAINER_NO_LIB -DBOOST_CONTAINER_STATIC_LINK -DFMT_HEADER_ONLY=1 -DTT_ENABLE_LIGHT_METAL_TRACE=1 -I/workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/ttnn/api -I/workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/ttnn/api/ttnn -I/workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/ttnn/cpp -I/workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/ttnn/. -I/workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/build/ttnn/flatbuffers -I/workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/tt_metal/api -I/workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/tt_metal/third_party/umd/device/api -I/workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/tt_metal/hostdevcommon/api -I/workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/tt_stl/. -isystem /workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/.cpmcache/magic_enum/4d76fe0a5b27a0e62d6c15976d02b33c54207096/include -isystem /workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/.cpmcache/fmt/69912fb6b71fcb1f7e5deca191a2bb4748c4e7b6/include -isystem /workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/core/include -isystem /workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/assert/include -isystem /workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/config/include -isystem /workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/static_assert/include -isystem /workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/throw_exception/include -isystem /workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/container/include -isystem /workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/intrusive/include -isystem /workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/move/include -isystem /workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/tt_metal/third_party/tracy/public -isystem /workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/.cpmcache/nlohmann_json/798e0374658476027d9723eeb67a262d0f3c8308/include -isystem /workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/.cpmcache/reflect/e75434c4c5f669e4a74e4d84e0a30d7249c1e66f -isystem /workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/.cpmcache/xtensor/0_26_0_patched/include -isystem /workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/.cpmcache/xtl/0_8_0_patched/include -isystem /workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/.cpmcache/flatbuffers/2c4062bffa52fa4157b1b4deeae73395df475fda/include -O0 -g -DDEBUG -std=c++20 -fPIC -fsized-deallocation -mavx2 -fno-lto -fPIC -fvisibility-inlines-hidden -Wall -Werror -Wconditional-uninitialized -Wno-deprecated-declarations -Wno-unused-function -Wno-unused-variable -Wno-c++11-narrowing -Wno-deprecated-volatile -Wno-for-loop-analysis -Wno-logical-op-parentheses -Wno-mismatched-tags -Wno-missing-braces -Wno-reorder-ctor -Wno-unused-lambda-capture -Wno-unused-local-typedef -Wno-unused-private-field -Wno-unknown-pragmas -Wno-int-to-pointer-cast -MD -MT ttnn/CMakeFiles/ttnn_core.dir/core/graph/graph_argument_serializer.cpp.o -MF ttnn/CMakeFiles/ttnn_core.dir/core/graph/graph_argument_serializer.cpp.o.d -o ttnn/CMakeFiles/ttnn_core.dir/core/graph/graph_argument_serializer.cpp.o -c /workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/ttnn/core/graph/graph_argument_serializer.cpp
/workspaces/oxpython/components/backends/tenstorrent/externals/tt-metal/ttnn/core/graph/graph_argument_serializer.cpp:9:10: fatal error: 'boost/algorithm/string/replace.hpp' file not found
    9 | #include <boost/algorithm/string/replace.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

Include path with `.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/algorithm/include` needs to be in the include directories list, but it's not.

I'm not doing anything fancy for the build, just running the build through `build_metal.sh`

CC: @afuller-TT 

### Steps to reproduce the issue

1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error


### Expected behavior

_No response_

### Please complete the following environment information

_No response_

--- a/ttnn/CMakeLists.txt
+++ b/ttnn/CMakeLists.txt
@@ -72,6 +72,7 @@ target_link_libraries(
         TT::Metalium
         xtensor
         FlatBuffers::FlatBuffers
+        Boost::algorithm
 )
 
 ##################################################

